packages/aws-cdk-lib/aws-apigateway/test/restapi.test.ts

```diff
@@ -915,6 +915,45 @@ describe('restapi', () => {
       });
     }).toThrow(/'cloudWatchRole' must be enabled for 'cloudWatchRoleRemovalPolicy' to be applied./);
   });
+
+  test('mode property is set correctly', () => {
+    // WHEN
+    const apiWithOverwrite = new apigw.RestApi(stack, 'api-overwrite', {
+      mode: apigw.RestApiMode.OVERWRITE,
+    });
+    apiWithOverwrite.root.addMethod('GET');
+
+    const apiWithMerge = new apigw.RestApi(stack, 'api-merge', {
+      mode: apigw.RestApiMode.MERGE,
+    });
+    apiWithMerge.root.addMethod('GET');
+
+    // THEN
+    Template.fromStack(stack).hasResourceProperties('AWS::ApiGateway::RestApi', {
+      Name: 'api-overwrite',
+      Mode: 'overwrite',
+    });
+
+    Template.fromStack(stack).hasResourceProperties('AWS::ApiGateway::RestApi', {
+      Name: 'api-merge',
+      Mode: 'merge',
+    });
+  });
+
+  test('mode property is optional', () => {
+    // WHEN
+    const api = new apigw.RestApi(stack, 'api');
+    api.root.addMethod('GET');
+
+    // THEN
+    Template.fromStack(stack).hasResourceProperties('AWS::ApiGateway::RestApi', {
+      Name: 'api',
+    });
+    // Mode should not be present in the template when not specified
+    Template.fromStack(stack).hasResourceProperties('AWS::ApiGateway::RestApi', {
+      Mode: Match.absent(),
+    });
+  });
```

Comment:
We can change like this:

```suggestion
  test.each([
    [apigw.RestApiMode.OVERWRITE, 'overwrite'],
    [apigw.RestApiMode.MERGE, 'merge'],
    [undefined, undefined],
  ])('mode property is set correctly (%s)', (mode, expectedMode) => {
    // WHEN
    const api = new apigw.RestApi(stack, 'api', {
      mode,
    });
    api.root.addMethod('GET');

    // THEN
    Template.fromStack(stack).hasResourceProperties('AWS::ApiGateway::RestApi', {
      Name: 'api',
      Mode: mode ? expectedMode : Match.absent(),
    });
  });
```

Reply:
Thanks for the suggestion!